tests/puppeteer/pages/commonPage.js

```diff
@@ -10,4 +10,55 @@ module.exports = class CommonPage {
   async goTo(URL) {
     await this.page.goto(URL);
   }
+
+  /**
+   * Wait for selector and click
+   * @param selector, element to check
+   * @return boolean, true if text exist, false if not
+   */
+  async waitForSelectorAndClick(selector) {
+    await this.page.waitForSelector(selector, {visible: true, timeout: 10000});
+    await this.page.click(selector)
+  }
+
+  /**
+   * Check text value
+   * @param selector, element to check
+   * @param textToCheckWith, text to check with
+   * @param parameter, parameter to use
+   * @return boolean, true when the result of parameter(textToCheckWith) is true, false if not
+   */
+  async checkTextValue(selector, textToCheckWith, parameter = 'equal') {
+    switch (parameter) {
+      case "equal":
+        await this.page.waitFor(selector);
+        await this.page.$eval(selector, el => el.innerText).then((text) => {
+          if (text.indexOf('\t') != -1) {
+            text = text.replace("\t", "");
+          }
+          expect(text.trim()).to.equal(textToCheckWith)
+        });
+        break;
+      case "contain":
+        await this.page.waitFor(selector);
+        await this.page.$eval(selector, el => el.innerText).then((text) => expect(text).to.contain(textToCheckWith));
+        break;
+    }
+  };
+
+  /**
+   * Check attribute value
+   * @param selector, element to check
+   * @param attribute, attribute to test
+   * @param textToCheckWith, text to check with
+   * @return boolean, true if text in attribute is equal to textToCheckWith, false if not
+   */
+  async checkAttributeValue(selector, attribute, textToCheckWith) {
+    await this.page.waitFor(selector);
+    let value = await this.page.evaluate((selector, attribute) => {
+      let elem = document.querySelector(selector);
+      return elem.getAttribute(attribute);
+    }, selector, attribute);
```

Comment:
Instead of this 
```js
    let value = await this.page.evaluate((selector, attribute) => {
      let elem = document.querySelector(selector);
      return elem.getAttribute(attribute);
    }, selector, attribute);
```
Maybe use this 
```js
    let value = await page.$eval(selector, elm => elm.getProperty(attribute).jsonValue());
```

Reply:
Yes, thank you @boubkerbribri 
`let value = await this.page.$eval(selector, (el, attribute) => el.getAttribute(attribute), attribute);`